<html>
<h1 align="center">
  <img align="center"; src="https://s3.invisionapp-cdn.com/storage.invisionapp.com/boards/files/183060432.png?x-amz-meta-iv=1&x-amz-meta-ck=cd20ea812f8ae161523111afa5aea5e8&AWSAccessKeyId=AKIAJFUMDU3L6GTLUDYA&Expires=1622505600&Signature=bA3DcaKL2fG4ZmCUtuv5fSzcrLE%3D"; width="100"; height="90">
    Plataforma: (Ej: Pepito)
</h1>

<h3 align="center",  style="background: yellow; color:black"> Importante: No trabajar aquí. Hacer copia de este archivo. </h3>
</html>

<html><h2 align="center", style="background: #c91f45; color:white">Datos principales</h2></html>

- **Desarrollador:** Señor pepe
- **Fecha del documento:** Marzo 2021

<html><h2 align="center", style="background: #c91f45; color:white">Información de la plataforma</h2></html>

Pepito es una OTT de USA  que ofrece servicio de internet, cable y telefonía. Además tiene contenido on demand para sus subscriptores.

- **Web oficial:** https://www.pepito.com/ (Necesaria VPN)
- **Packages:** free-vod, subscription-vod.

<html><h2 align="center", style="background: #c91f45; color:white">Información del script</h2></html>

- **Librerías:** Usa Bs4, Selenium, Api, etc.
- **URL API:** https://api.pepito.com/v3/
- **URL BS4:** https://www.pepito.com/movies/, https://www.pepito.com/movies/
- **URL Selenium:** https://www.pepito.com/movies/, https://www.pepito.com/movies/
- Otros datos relevantes...

<html><h2 align="center", style="background: #c91f45; color:white">Información por país</h2></html>

**ar.pepito:**
- VPN para correr: No
- VPN para ver: No
- Cantidad Contenidos: 10 peliculas. 10 series y 50 capítulos.
- Tiempo de ejecución: dos días ahre.

**us.pepito:**
- VPN para correr: VPN en US. ExpressVPN.
- VPN para ver: VPN en US. ExpressVPN.
- Cantidad Contenidos: 10 peliculas. 10 series y 50 capítulos.
- Tiempo de ejecución: dos días ahre.

**dz.pepito:**
- VPN para correr: No
- VPN para ver: No
- Cantidad Contenidos: 10 peliculas. 10 series y 50 capítulos.
- Tiempo de ejecución: dos días ahre.

<html><h2 align="center", style="background: #c91f45; color:white">Otros datos importantes</h2></html>

Es importante tener en cuenta...

<h1 align="center"> Índice del análisis </h1>

<html><h2 align="center", style="background: #8f1b4d; color:white">titanScraping</h2></html>

- [**Inicio del análisis**](#Inicio-del-análisis)
- [**Id**](#Id)
- [**Title y CleanTitle**](#Title-y-CleanTitle)
- [**Duration**](#Duration)
- [**Rating y Genres**](#Rating-y-Genres)
- [**Deeplinks e Image**](#Deeplinks-e-Image)
- [**Packages**](#Packages)
- [**Synopsis**](#Synopsis)
- [**IsOriginal, IsBranded y Provider**](#IsOriginal,-IsBranded-y-Provider)
- [**Cast y Directors**](#Cast-y-Directors)
- [**Year y Country**](#Year-y-Country)
- [**Resto de datos**](#Resto-de-datos)

<html><h2 align="center", style="background: #0C2243; color:white">titanScrapingEpisodes</h2></html>

- [**Análisis de Episodios**](#Análisis-de-Episodios)
- [**Id y Title**](#Id-y-Title)
- [**Series**](#Series)
- [**Episode y Season**](#Episode-y-Season)
- [**Deeplinks e Image (Episodios)**](#Deeplinks-e-Image-(Episodios))
- [**Resto de datos (Episodios)**](#Resto-de-datos-(Episodios))

<html><h2 align="center", style="background: #72287F; color: white">apiPresence</h2></html>

- [**Compración y análisis**](#Comparación-y-análisis)

<html><h2 align="center", style="background: #90E0B0; color: black">Conclusiones</h2></html>

- [**Ir a las conclusiones**](#Conclusiones:)

<html><h1 align="center", style="background: #8f1b4d; color:white">titanScraping</h1></html>

## Inicio del análisis

- Traer DataFrame de **titanScraping**
- Setear el formato de los datos.
- Contar Cantidades
- Datos que parezcan relevantes

**Funciones Recomendadas:**

- df.info()
- df.shape
- **utils.GetDataFrame** -> Para traer DataFrame
- **utils.PayloadAnalysis**.***firtst_check(DataFrame)*** -> Para hacer un análisis inicial del DataFrame.

In [1]:
# Genero el DataFrame:
from utils import GetDataFrame
query = {"PlatformCode":"ar.starzplay"}
df = GetDataFrame.local(query)
df


 --- Conectando a localhost --- 
 --- BBDD -> titan.
 --- Collection -> titanScraping. (Por default)


_id  PlatformCode     Id  \
0    61264d9e9fec287f9767c1bb  ar.starzplay   5972   
1    61264d9e9fec287f9767c1bc  ar.starzplay   6612   
2    61264d9e9fec287f9767c1bd  ar.starzplay  16830   
3    61264d9e9fec287f9767c1be  ar.starzplay  18124   
4    61264d9e9fec287f9767c1bf  ar.starzplay  19568   
..                        ...           ...    ...   
212  61264d9e9fec287f9767c28f  ar.starzplay  63541   
213  61264d9e9fec287f9767c290  ar.starzplay  63543   
214  61264d9e9fec287f9767c291  ar.starzplay  63571   
215  61264d9e9fec287f9767c292  ar.starzplay  63736   
216  61264d9e9fec287f9767c293  ar.starzplay  63740   

                            Title OriginalTitle                    CleanTitle  \
0                       Spartacus          None                     Spartacus   
1                            Dune          None                          Dune   
2        No todo es lo que parece          None      No todo es lo que parece   
3                 The White Queen          None               The White Queen   
4                     Black Sails          None                   Black Sails   
..                            ...           ...                           ...   
212                      Ranchero          None                      Ranchero   
213               Posdata, te amo          None               Posdata, te amo   
214                         Reign          None                         Reign   
215  Baby: El aprendiz del crimen          None  Baby: El aprendiz del crimen   
216             Blade Runner 2049          None             Blade Runner 2049   

      Type  Year  Duration ExternalIds  ...  \
0    serie  2010       NaN        None  ...   
1    movie  1984     136.0        None  ...   
2    movie  2004     105.0        None  ...   
3    serie  2013       NaN        None  ...   
4    serie  2014       NaN        None  ...   
..     ...   ...       ...         ...  ...   
212  movie  2011      94.0        None  ...   
213  movie  2007     125.0        None  ...   
214  serie  2013       NaN        None  ...   
215  movie  2017     112.0        None  ...   
216  movie  2017     163.0        None  ...   

                                                  Cast  \
0    [Andy Whitfield, John Hannah, Liam McIntyre, L...   
1    [Kyle MacLachlan, Sting, Francesca Annis, Leon...   
2    [Daniel Craig, Colm Meaney, Kenneth Cranham, G...   
3    [Rebecca Ferguson, Aneurin Barnard, Amanda Hal...   
4    [Hannah New, Toby Stephens, Luke Arnold, Jessi...   
..                                                 ...   
212  [Roger Gutierrez, Brian Eric Johnson, Christin...   
213  [Anna Kendrick, Zoë Chao, Peter Brown, Sasha C...   
214  [Adelaide Kane, Megan Follows, Torrance Coombs...   
215  [Ansel Elgort, Kevin Spacey, Lily James, Jon B...   
216  [Ryan Gosling, Harrison Ford, Ana de Armas, Sy...   

                                             Directors Availability Download  \
0                                                 None         None     None   
1                                        [David Lynch]         None     None   
2                                     [Matthew Vaughn]         None     None   
3                                                 None         None     None   
4                                                 None         None     None   
..                                                 ...          ...      ...   
212                                  [Richard Kaponas]         None     None   
213                                               None         None     None   
214  [Allan Kroeker, Anne Wheeler, Bradley Walsh, B...         None     None   
215                                     [Edgar Wright]         None     None   
216                                 [Denis Villeneuve]         None     None   

    IsOriginal IsAdult                        Packages Country  \
0         None    None  [{'Type': 'subscription-vod'}]    None   
1         None    None  [{'Type': 'subscription-v

In [7]:
# Importo los módulos de análisis:
from utils import PayloadAnalysis, PlatformAnalysis, MoreAnalysis

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 217 entries, 0 to 216
Data columns (total 26 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   _id            217 non-null    object 
 1   PlatformCode   217 non-null    object 
 2   Id             217 non-null    object 
 3   Title          217 non-null    object 
 4   OriginalTitle  0 non-null      object 
 5   CleanTitle     217 non-null    object 
 6   Type           217 non-null    object 
 7   Year           217 non-null    int64  
 8   Duration       148 non-null    float64
 9   ExternalIds    0 non-null      object 
 10  Deeplinks      217 non-null    object 
 11  Synopsis       217 non-null    object 
 12  Image          217 non-null    object 
 13  Rating         217 non-null    object 
 14  Provider       0 non-null      object 
 15  Genres         217 non-null    object 
 16  Cast           211 non-null    object 
 17  Directors      179 non-null    object 
 18  Availabili

In [5]:
df.shape[0]

217

In [9]:
PayloadAnalysis.first_check(df)

 ***** Keys del Payload ***** 

¡Trae todos los valores (keys) del Payload okay!


**Primeras impresiones:**

Todo ok por el momento...

<html><h2 align="center", style="background: #8f1b4d; color:white">Id</h2></html>

## Id

- Verificar que sean únicos.
- Que no cambien en el tiempo.
- Evaluar usar hashlib.

**Funciones Recomendadas:**
- df['Id'].nunique()
- **utils.PayloadAnalysis**.***check_id(DataFrame)*** -> Verificar ids.

In [22]:
df['Id'].nunique() == df.shape[0]

True

In [28]:
for content in df['Deeplinks']:
        print(content['Web'])

https://www.starz.com/ar/es/series/5972
https://www.starz.com/ar/es/movies/6612
https://www.starz.com/ar/es/movies/16830
https://www.starz.com/ar/es/series/18124
https://www.starz.com/ar/es/series/19568
https://www.starz.com/ar/es/series/21220
https://www.starz.com/ar/es/movies/22895
https://www.starz.com/ar/es/movies/23866
https://www.starz.com/ar/es/movies/23922
https://www.starz.com/ar/es/movies/24244
https://www.starz.com/ar/es/series/24528
https://www.starz.com/ar/es/movies/25250
https://www.starz.com/ar/es/series/25586
https://www.starz.com/ar/es/movies/25826
https://www.starz.com/ar/es/movies/26226
https://www.starz.com/ar/es/movies/26914
https://www.starz.com/ar/es/movies/28004
https://www.starz.com/ar/es/movies/29251
https://www.starz.com/ar/es/movies/29285
https://www.starz.com/ar/es/movies/29666
https://www.starz.com/ar/es/movies/30168
https://www.starz.com/ar/es/movies/30233
https://www.starz.com/ar/es/movies/30262
https://www.starz.com/ar/es/movies/30698
https://www.starz.

**Conclusiones Id:**

Trae bien los ids porque tiene una cantidad de 217 ids unibocas y hay 217 contenidos.

<html><h2 align="center", style="background: #8f1b4d; color:white">Title y CleanTitle</h2></html>

## Title y CleanTitle

- Verificar títulos duplicados.
- Verificar si los CleanTitles efectivamente están limpios.
- Mirar el **punto 2** en [ClickUp](https://app.clickup.com/3043480/docs/2ww4r-192/2ww4r-388)

**Funciones Recomendadas:**
- **utils.PayloadAnalysis**.***check_title(DataFrame)*** -> Análisis de Title.
- **utils.PayloadAnalysis**.***check_clean_title(DataFrame)*** -> Análisis de CleanTitle.

In [23]:
PayloadAnalysis.check_title(df)

 ***** Análisis Title ***** 

 ##### DUPLICADOS: ##### 

DUPLICADOS: NO HAY DUPLICADOS 


 ####################### 



**Conclusiones Titles y CleanTitles:**

No hay contenido duplicado, pero bla bla bla...

<html><h2 align="center", style="background: #8f1b4d; color:white">Duration</h2></html>

## Duration

- Verificar si la duración está bien, si nada nos llama la atención.
- Dudar de los títulos de menos de 10 minutos o los contenidos que duren mucho. (Puede haber trailers)

**Funciones Recomendadas:**
- **utils.PayloadAnalysis**.***check_duration(DataFrame)*** -> Análisis de Duration.

**Conclusiones Duration:**

Bla bla bla..

In [ ]:
<html><h2 align="center", style="background: #8f1b4d; color:white">Rating y Genres</h2></html>

## Rating y Genres

- Verificamos cuantos tipos de Ratings hay.
- Verificar que sean efectivamente Ratings.
- Verificar que los tipos de Genres y que efectivamente sea un Genre. (Un año no es un Genre)

**Funciones Recomendadas:**
- df['Rating'].unique()
- **utils.PayloadAnalysis**.***check_rating(DataFrame)*** -> Análisis de Rating.
- **utils.PayloadAnalysis**.***check_genres(DataFrame)*** -> Análisis de Genres.

**Conclusiones Rating y Genres:**

En este caso..

<html><h2 align="center", style="background: #8f1b4d; color:white">Deeplinks e Image</h2></html>

## Deeplinks e Image

- Muestra aleatoria para verificar **Deeplink e Images.**
- Verificar que ambos funcionen bien.

**Funciones Recomendadas:**
- **utils.PayloadAnalysis**.***check_deeplinks(DataFrame)*** -> Análisis de Deeplinks.
- **utils.PayloadAnalysis**.***check_image(DataFrame)*** -> Análisis de Image.

**Conclusiones Deeplinks e Image:**

Las trae bien o mal???

<html><h2 align="center", style="background: #8f1b4d; color:white">Packages</h2></html>

## Packages

- Revisar detalladamente que los packages estén bien.
- Como me doy cuenta el tipo de package, ¿Hay tag o key del API que indica el package?
- Ver si coincide con los package que indica PlatformValidation.

**Funciones Recomendadas:**
- **utils.PayloadAnalysis**.***check_packages(DataFrame)*** -> Análisis de los Packages.

**Conclusiones Packages:**

Estan bien o faltan.

<html><h2 align="center", style="background: #8f1b4d; color:white">Synopsis</h2></html>

## Synopsis

- Revisar que efectivamente sean Synopsis.
- Revisar que la synopsis venga sin tags embebidos ni carecteres raros o sin correcta codificación.

**Funciones Recomendadas:**
- **utils.PayloadAnalysis**.***check_synopsis(DataFrame)*** -> Análisis de Synopsis.

**Conclusiones de Synopsis:**

Bla bla bla...

<html><h2 align="center", style="background: #8f1b4d; color:white">IsOriginal, IsBranded y Provider</h2></html>

## IsOriginal, IsBranded y Provider

- Revisar bien en caso de traer Providers. Por ejemplo acá puede ir un channel. Ej: Amazon tiene a Starzplay como channel.
- Validar si hay Branding en la plataforma con el campo IsBranded.
- Identificar si el contenido es original de la plataforma con IsOriginal.

**Funciones Recomendadas:**
- **utils.PayloadAnalysis**.***check_is_original(DataFrame)*** -> Análisis de IsOriginal.
- **utils.PayloadAnalysis**.***check_is_branded(DataFrame)*** -> Análisis de IsBranded.
- **utils.PayloadAnalysis**.***check_provider(DataFrame)*** -> Análisis de Provider.

**Conclusiones IsOriginal, IsBranded y Providers:**

Estan bien o faltan.

<html><h2 align="center", style="background: #8f1b4d; color:white">Cast y Directors</h2></html>

## Cast y Directors

- Revisar que efectivamente sean Cast y Directors.
- Revisar que no haya "()" o apodos en Cast o Directors.

**Funciones Recomendadas:**
- **utils.PayloadAnalysis**.***check_cast(DataFrame)*** -> Análisis de Cast.
- **utils.PayloadAnalysis**.***check_directors(DataFrame)*** -> Análisis de Directors.

**Conclusiones Cast y Directors:**

Bla Bla bla...

<html><h2 align="center", style="background: #8f1b4d; color:white">Year y Country</h2></html>

## Year y Country

- Verificar que los años estén bien, que no tire un 2037 (?.
- Verificar que los Countries sean del país el cual produjo el contenido.

**Funciones Recomendadas:**
- **utils.PayloadAnalysis**.***check_year(DataFrame)*** -> Análisis de Year.
- **utils.PayloadAnalysis**.***check_country(DataFrame)*** -> Análisis de Country.

**Conclusiones Year y Country:**

Bla Bla bla...

<html><h2 align="center", style="background: #8f1b4d; color:white">Resto de datos</h2></html>

## Resto de datos

- Revisar si se consiguieron los campos Availability, Download, Seasons, IsAdult, ExternalIds, Etc.

<html><h3 align="center",  style="background: yellow; color:black"> Importante: Si hay ExternalIds con info de IMDB, traer este campo sí o si. </h3></html>


**Funciones Recomendadas:**
- **utils.PayloadAnalysis**.***check_availability(DataFrame)*** -> Análisis de Availability.
- **utils.PayloadAnalysis**.***check_download(DataFrame)*** -> Análisis de Download.
- **utils.PayloadAnalysis**.***check_seasons(DataFrame)*** -> Análisis de Seasons.
- **utils.PayloadAnalysis**.***check_externalids(DataFrame)*** -> Análisis de ExternalIds.

**Conclusiones de Resto de los datos:**

Bla Bla bla...

<html><h2 align="center", style="background: #0C2243; color:white">titanScrapingEpisodes</h2></html>

## Análisis de Episodios

- Traer DataFrame de **titanScrapingEpisodes**
- Setear el formato de los datos.
- Contar Cantidades
- Datos que parezcan relevantes

**Funciones Recomendadas:**
 - df.info()
 - df.shape
 - **utils.EpisodesAnalysis**.***check_episodes(DataFrame)*** -> Análisis de inicial de los Episodios.

In [1]:
# Genero el DataFrame de los episodios:
query = {"PlatformCode":"us.pepito"}
df_epi = GetDataFrame.local(query, collection='titanScrapingEpisodes')
df_epi

NameError: name 'GetDataFrame' is not defined

In [ ]:
# Importo los módulos de análisis de episodios:
from utils import EpisodesAnalysis, MoreAnalysis

<html><h2 align="center", style="background: #0C2243; color:white">Id y Title</h2></html>

## Id y Title

- ¿Se repiten Ids o títulos?

**Funciones Recomendadas:**
 - df.info()
 - **utils.EpisodesAnalysis**.***check_id(DataFrame)*** -> Análisis de ids de Episodios.
 - **utils.EpisodesAnalysis**.***check_title(DataFrame)*** -> Análisis de Titles de Episodios.

<html><h2 align="center", style="background: #0C2243; color:white">Series</h2></html>

## Series

- Si las series está homogeneizadas.
- Verificar si hay temporadas especiales.
- ¿Traen todas las temporadas disponibles?
- ¿Hay series sin episodios?
- ¿Trae todos los números de episodios? ¿Trae todos los números de temporadas?

**Funciones Recomendadas:**
 - **utils.EpisodesAnalysis**.***check_series(DataFrame)*** -> Análisis de las Series.

**Conclusión Series:**

En este caso..

<html><h2 align="center", style="background: #0C2243; color:white">Episode y Season</h2></html>

## Episode y Season

- ¿Trea todos los números de Episodios?
- ¿Trae todos los números de Temporadas?

**Funciones Recomendadas:**
 - **utils.EpisodesAnalysis**.***check_episode(DataFrame)*** -> Análisis de número de Episodios.
 - **utils.EpisodesAnalysis**.***check_season(DataFrame)*** -> Análisis de número de Season.

<html><h2 align="center", style="background: #0C2243; color:white">Deeplinks e Image (Episodios)</h2></html>

## Deeplinks e Image (Episodios)

- Testear Deeplinks e Images de los episodios.

**Funciones Recomendadas:**
 - **utils.EpisodesAnalysis**.***check_deeplinks(DataFrame)*** -> Análisis de deeplinks.
 - **utils.EpisodesAnalysis**.***check_image(DataFrame)*** -> Análisis de image.

<html><h2 align="center", style="background: #0C2243; color:white">Resto de datos (Episodios)</h2></html>

## Resto de datos (Episodios)

- Verificar si los episodios traen más datos, como synopsis, etc.

**Funciones Recomendadas:**
 - **utils.EpisodesAnalysis**.***check_synopsis(DataFrame)*** -> Análisis de synopsis de episodios.

<html><h1 align="center", style="background: #72287F; color: white">apiPresence</h1></html>
</html>

## Comparación y análisis

- Se puede traer DataFrame de **apiPresence** y realizar comparaciones (Si corresponde).
- Se debe pedir el .csv de apiPresence al equipo. 


In [ ]:
query = {"PlatformCode":"us.pepito", "CreatedAt": "2019-01-01"}
PlatformAnalysis.compare_apiPresence(query) # Compara mongo local con apiPresence.

<html><h1 align="center", style="background: #90E0B0; color :black">Conclusiones:</h1></html>

- Trae 12.xxx contenidos y 54.xxx episodios.
- Titles y CleanTitles vienen ...
- Trae todos los ratings...
- Trae todos los packages...
- Otros datos importantes...

## Pendiente a averiguar: (Si corresponde)
- Le podríamos pedir a Platform Validation que investigue...
- Me quedé con dudas en...